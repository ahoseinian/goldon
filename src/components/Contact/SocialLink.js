import React from 'react'
import styled from 'styled-components'
import {color} from '../theme'
import Icon from '../common/Icon'

const LinkButton = styled.a`
  color:${color.dark}
`

const SocialLink = ({url, label}) => (
  <LinkButton href={url} target="blank" className="mx-2">
    <Icon name={label}/>
  </LinkButton>
)

SocialLink.propTypes = {
  url: React.PropTypes.string.isRequired,
  label: React.PropTypes.string.isRequired,
}

export default SocialLink
