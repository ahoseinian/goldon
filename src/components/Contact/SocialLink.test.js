import React from 'react'
import {shallow} from 'enzyme'
import SocialLink from './SocialLink'
import toJson from 'enzyme-to-json';

const item = {
  url: 'https://t.me/cactusang',
  label: 'telegram'
}

describe('SocialLink', () => {
  it('renders', () => {
    const wrapper = shallow(<SocialLink {...item}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
