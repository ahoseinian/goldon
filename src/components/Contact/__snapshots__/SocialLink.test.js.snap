exports[`SocialLink renders 1`] = `
<styled.a
  className="mx-2"
  href="https://t.me/cactusang"
  target="blank">
  <IconSpan
    name="telegram" />
</styled.a>
`;
